app/src/main/java/com/hyperether/getgoing/activity/ShowDataActivity.java

```diff
@@ -61,6 +73,98 @@ protected void onListItemClick(ListView l, View v, int position, long id) {
+        ArrayList<String> charDates = new ArrayList<>();
+        for (int i = 0; i < routes.size(); i++) {
+            charDatesSet.add(routes.get(i).getDate().substring(0, 10));
+        }
+        charDates.addAll(charDatesSet);
+        return charDates;
+    }
+
+    private ArrayList<Map<String, Float>> collectData() {
+        ArrayList<Map<String, Float>> collectedData = new ArrayList<>();
+        Map<String, Float> mapY1 = new LinkedHashMap<>();
+        Map<String, Float> mapY2 = new LinkedHashMap<>();
+        Map<String, Float> mapY3 = new LinkedHashMap<>();
+
+        for (DbRoute item : routes) {
+            if (mapY1.containsKey(item.getDate().substring(0, 10)) && item.getActivity_id() == 1) {
+                float q = mapY1.get(item.getDate().substring(0, 10)) + (float) item.getEnergy();
+                mapY1.put(item.getDate().substring(0, 10), q);
+            } else if (item.getActivity_id() == 1) {
+                mapY1.put(item.getDate().substring(0, 10), (float) item.getEnergy());
+            } else if (mapY2.containsKey(item.getDate().substring(0, 10)) &&
+                    item.getActivity_id() == 2) {
+                float q = mapY2.get(item.getDate().substring(0, 10)) + (float) item.getEnergy();
+                mapY2.put(item.getDate().substring(0, 10), q);
+            } else if (item.getActivity_id() == 2) {
+                mapY2.put(item.getDate().substring(0, 10), (float) item.getEnergy());
+            } else if (mapY3.containsKey(item.getDate().substring(0, 10)) &&
+                    item.getActivity_id() == 3) {
+                float q = mapY3.get(item.getDate().substring(0, 10)) + (float) item.getEnergy();
+                mapY3.put(item.getDate().substring(0, 10), q);
+            } else if (item.getActivity_id() == 3) {
+                mapY3.put(item.getDate().substring(0, 10), (float) item.getEnergy());
+            }
+        }
+        collectedData.add(mapY1);
+        collectedData.add(mapY2);
+        collectedData.add(mapY3);
+        return collectedData;
+    }
+
+    private ArrayList<IBarDataSet> prepareData() {
+        ArrayList<IBarDataSet> dataSets = new ArrayList<>();
+        ArrayList<BarEntry> yVal1 = new ArrayList<>();
+        ArrayList<BarEntry> yVal2 = new ArrayList<>();
+        ArrayList<BarEntry> yVal3 = new ArrayList<>();
+        ArrayList<String> x = xValuesChart();
+        Map<String, Float> y1 = collectData().get(0);
+        Map<String, Float> y2 = collectData().get(1);
+        Map<String, Float> y3 = collectData().get(2);
+
+        for (int i = 0; i < x.size(); i++) {
+            String xVal = x.get(i);
+            if (y1.containsKey(xVal)) {
+                yVal1.add(new BarEntry(y1.get(xVal), i));
+            }
+            if (y2.containsKey(xVal)) {
+                yVal2.add(new BarEntry(y2.get(xVal), i));
+            }
+            if (y3.containsKey(xVal)) {
+                yVal3.add(new BarEntry(y3.get(xVal), i));
+            }
+        }
+
```

Comment:
@NikolaHadzicNs can we add default value for a day for a cloumn, which is zero or some really small number, just to show all 3 graphs per day? Currently it looks like something is missing here, not intuitive that empty space is meant for other types of movement. Check situtation when yVal2 is empty after this piece of code..